### DISCUSSION TITLE
production builds and `prerender-manifest.json`

### DISCUSSION THREAD

**Discussion by josefaidt** [vercel/next.js#10668] [Help] [answered]
I'm having an issue with my Next.js site where I'm building the site with `next build`, and when attempting to run the site using my production environment vars and custom server with `NODE_ENV=production node server.js` I'm experiencing two different things:

1. locally, I can build the site and run it just fine. No issues.
2. after deploying to target host, the site fails to run with NODE_ENV set to `production`, as it is looking for the `prerender-manifest.json` in the build directory `/.next`, `packages/app/.next/prerender-manifest.json`

<details>
<summary>Stack Trace</summary>

```
fs.js:114
    throw err;
    ^

Error: ENOENT: no such file or directory, open '/home/some-user/packages/app/.next/prerender-manifest.json'
at Object.openSync (fs.js:443:3)
    at Object.readFileSync (fs.js:343:35)
    at Object.initializeSprCache (/home/some-user/packages/app/node_modules/next/dist/next-server/server/spr-cache.js:53:53)
    at new Server (/home/some-user/packages/app/node_modules/next/dist/next-server/server/next-server.js:92:21)
    at createServer (/home/some-user/packages/app/node_modules/next/dist/server/next.js:2:133)
    at Object.<anonymous> (/home/some-user/packages/app/server.js:19:13)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
error Command failed with exit code 1.
```
</details>

This doesn't make any sense to me because the file doesn't exist locally either, but I have no issues running the app.

### Deploy procedure:

1. build the app `next build`
2. copy app dir (minus `node_modules`) to target host
3. install deps with `npm install`/`yarn install`
4. run the app with `NODE_ENV=production node server.js`
  1. **NOTE** running the app with `NODE_ENV=staging node server.js` produces no issues, as `prerender-manifest.json` is not used in dev

Following this build procedure locally produces no errors.

## Question

Is there an error in my deploy or build procedure? I don't have any issues outside this one error. The reason I build locally and copy is the `next` builder is attempting to use Git to check modified time of MDX files and Git is not/will not be installed on the target host

**Comment by Timer** [comment]
It's impossible for us to provide assistance without a full reproduction provided.

Please provide a full reproducing (preferably minimal) example so we can identify the bug in your setup!

---

One thing that comes to mind is you may have a Next.js version mismatch between your build & deployment server. Be sure you're copying `yarn.lock` or `package-lock.json` to your production server.

**Comment by josefaidt** [comment]
@Timer I'll try to get something together as this is a private repo. In the meantime, should I be seeing a `prerender-manifest.json` file in the `.next` build dir locally? 
![image](https://user-images.githubusercontent.com/5033303/75186489-938e7680-570d-11ea-94ff-5adf5079f65f.png)

**Comment by Timer** [comment, marked as answer]
It depends, if you are not seeing a `prerender-manifest.json` in that directory locally, but the production server is looking for one, it's likely you have a Next.js version mismatch.

Only new Next.js versions emit the `prerender-manifest.json` file.

Be sure you're copying `yarn.lock` or `package-lock.json` to your production server **before** running `yarn install` or `npm install`.

**Reply by btabur** [reply]
![image](https://github.com/user-attachments/assets/0014b20e-63dc-45a8-8414-f70ca4d3c75e)
Should I copy the package-lock.json file to the server folder here?

**Comment by josefaidt** [comment]
Interestingly enough I had `9.2.0` in the local lockfile and `9.2.2` on the server. I was originally not copying the lockfile since it was in a monorepo, but I copied the root monorepo lockfile and was able to install the right version.

Just like that everything works :)

**Reply by Timer** [reply]
Great! Glad it's working now.

**Comment by dreamyguy** [comment]
By now I have google'd enough to know that this is a "classic" misconfiguration problem. But knowing that doesn't solve the problem...

There are no version mismatches on my case, I have the same Next version locally and remotely, Netlify's log tells me as much.

Is there a list of possible causes for this error somewhere in the docs? I can eventually update the docs with a PR, but for now I'm just stuck with this, and this issue is currently the top Google result for:

```
Error: .next/prerender-manifest.json: ENOENT: no such file or directory, open '.next/prerender-manifest.json'
```

**Reply by jinnkhan88** [reply]
@dreamyguy did you find any solution to this ? I am having the same issue on nextjs with amplify

**Reply by LyncLab-Engineering** [reply]
I had the exact same issue with amplify too, there isn't mismatch of next version in my case. Please update if you had found the solution, appreciate it much

**Comment by alymohamedhassan** [comment]
Hello there, I use Next.js v11.1.2, and there is no version difference bet. my local and server version. however this issue only exists on the server.

Any Solutions??

**Reply by pswai** [reply]
@alymohamedhassan It could be due to https://nextjs.org/docs/messages/prerender-error. For our case we accidentally put non-page components in the `pages` folder.

**Reply by alymohamedhassan** [reply]
I made sure that none of these issues existed.
I also need to understand why the server ask me for the prerender-manifest.json file

**Comment by adi-igit** [comment]
The issue was that my pages was generating not correctly due my path to pages wasn't correct. By fixing path to pages nextjs built all pages and then i could run build version. Also try checking that your pages collecting and accessing data correctly if you have any data included.

**Comment by MartinPavlik** [comment]
I just had the same problem and the reason why I was getting the error about missing `prerendered-manifest.json` was that one of the `getStaticProps` calls threw this error:
```
Error occurred prerendering page "/blog/xxx". Read more: https://nextjs.org/docs/messages/prerender-error
TypeError: Cannot read properties of undefined (reading 'some')
```

**Reply by ahmedivy** [reply]
So how you solved that, in my case i am generating 1000s of static pages and some pages often throw errors

**Reply by adityagoel-hexgit** [reply]
Same for me. getStaticProps threw an error. Once it got fixed, everything worked fine.

**Comment by nousidis** [comment]
Removing node modules and reinstalling worked for me. Was having this issue with Next 14.0.4.

**Reply by zariyatyasin** [reply]
Hello i'm having the same issues in next js 13.4. but i'm not undersetting where should i fix. 

no such file or directory, open '\\?\E:\Programming\Ecomerce\sudzar\.next\prerender-manifest.json'

**Comment by zariyatyasin** [comment]
i'm also having the same issues, after npm run build when i try start the application by npm start. it give me this error. pls anyone help me.

 no such file or directory, open '\\?\E:\Programming\Ecomerce\sudzar\.next\prerender-manifest.json'

![3434](https://github.com/vercel/next.js/assets/64485419/663ed911-30e6-4112-a649-19e631674636)

**Reply by MartinPavlik** [reply]
Can you provide the output of the build command?

**Comment by Octavian-imp** [comment]
I found another solution. I thought that if App Router is selected, then the pages folder can be used as components for pages, but it turns out Next did not provide for this and in my case it caused an error.

**Comment by CjordanIII** [comment]
Solution don't use next

**Reply by Octavian-imp** [reply]
Why?

**Comment by gayatritondare48** [comment]
My comment from Postman

**Comment by gayatritondare48** [comment]
Comment from AutomationEdge

**Comment by gayatritondare48** [comment]
Comment from AutomationEdge
